apps/server/src/form-instances/form-instances.service.ts

```diff
@@ -300,6 +300,42 @@ export class FormInstancesService {
       ...updatedSignature,
     };
 
-    return formInstance;
+    const allSigned = formInstance.signatures.every((sig) => sig.signed);
+
+    if (allSigned) {
+      await this.prisma.formInstance.update({
+        where: { id: formInstanceId },
+        data: { completed: true, completedAt: new Date() },
+      });
+    }
+
+    const updatedFormInstance = await this.findOne(formInstanceId);
```

Comment:
the update method should return the updated form instance, so we wouldn't have to requery

Reply:
i needed to make this query so the updated form instance would also reflect the updated signatures regardless of allSigned. I cleaned the code up a little in the latest commit